packages/store/addon/-private/ts-interfaces/minimum-serializer-interface.ts

```diff
@@ -0,0 +1,280 @@
+import ShimModelClass from '../system/model/shim-model-class';
+
+type OptionsHash = Record<string, any>;
+
+interface Serializer {
+  /**
+   * This method is responsible for normalizing the value resolved from the promise returned
+   * by an Adapter request into the format expected by the `Store`.
+   *
+   * The output should be a [JSON:API](https://jsonapi.org/) document with the following
+   * additional restrictions:
+   *
+   * - `type` should be formatted in the `singular` `dasherized` `lowercase` form
+   * - `members` (the property names of attributes and relationships) should be formatted
+   *    to match their definition in the corresponding `Model` definition. Typically this
+   *    will be `camelCase`.
+   *
+   * @method normalizeResponse
+   * @public
+   * @param {Store} store - the store service that initiated the request being normalized
+   * @param {ShimModelClass} schema - An object with methods for accessing information about
+   *  the type, attributes and relationships of the primary type associated with the request.
+   * @param {JSONObject} rawPayload - The raw JSON response data returned from an API request.
+   *  This correlates to the value the promise returned by the adapter method that performed
+   *  the request resolved to.
+   * @param {string|null} id - For a `findRecord` request, this is the `id` initially provided
+   *  in the call to `store.findRecord`. Else this value is `null`.
+   * @param {'findRecord' | 'queryRecord' | 'findAll' | 'findBelongsTo' | 'findHasMany' | 'findMany' | 'query' | 'createRecord' | 'deleteRecord' | 'updateRecord'} requestType - The
+   *  type of request the Adapter had been asked to perform.
+   *
+   * @returns {JsonApiDocument} - a document following the structure of a `JSON:API` Document.
+   */
+  normalizeResponse(
+    store: Store,
+    schema: ShimModelClass,
+    rawPayload: JSONObject,
+    id: string | null,
+    requestType:
+      | 'findRecord'
+      | 'queryRecord'
+      | 'findAll'
+      | 'findBelongsTo'
+      | 'findHasMany'
+      | 'findMany'
+      | 'query'
+      | 'createRecord'
+      | 'deleteRecord'
+      | 'updateRecord'
+  ): JsonApiDocument;
+
+  /**
+   * This method is responsible for serializing an individual record
+   * via a [Snapshot](Snapshot) into the format expected by the API.
+   *
+   * This method is called by `snapshot.serialize()`.
+   *
+   * When using `Model`, this method is called by `record.serialize()`.
+   *
+   * When using `JSONAPIAdapter` or `RESTAdapter` this method is called
+   * by `updateRecord` and `createRecord` if `Serializer.serializeIntoHash`
+   * is not implemented.
+   *
+   * @method serialize
+   * @public
+   * @param {Snapshot} snapshot - A Snapshot for the record to serialize
+   * @param {object} options
+   */
+  serialize(snapshot: Snapshot, options?: OptionsHash): JSONObject;
+
+  /**
+   * This method is intended to normalize data into a `JsonApiDocument` representing
```

Comment:
lets add a link to the specific part of the JSON:API spec that this is talking about.